arangod/Agency/Store.cpp

```diff
@@ -358,21 +358,28 @@ std::vector<bool> Store::applyLogEntries(arangodb::velocypack::Builder const& qu
       if (endpointPathFromUrl(url, endpoint, path)) {
 
         Agent* agent = _agent;
-        network::sendRequest(cp, endpoint, fuerte::RestVerb::Post, path,
-                             *buffer, reqOpts).thenValue([=](network::Response r) {
-          if (r.fail() || r.response->statusCode() >= 400) {
-            LOG_TOPIC("9dbf0", TRACE, Logger::AGENCY)
-              << url << "(" << r.response->statusCode() << ", " << fuerte::to_string(r.error)
-              << "): " << r.slice().toJson();
-
-            if (r.ok() && r.response->statusCode() == 404 && _agent != nullptr) {
-              LOG_TOPIC("9dbfa", DEBUG, Logger::AGENCY) << "dropping dead callback at " << url;
-              agent->trashStoreCallback(url, VPackSlice(buffer->data()));
-            }
-          }
-
-        });
+        try {
+          network::sendRequest(cp, endpoint, fuerte::RestVerb::Post, path, *buffer, reqOpts).thenValue(
```

Comment:
is this actually throwing? Its really not supposed to

Reply:
```

#0  0x00007ffff7a50762 in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x00000000015f4c57 in arangodb::fuerte::v1::ConnectionBuilder::endpoint (this=<optimized out>, Python Exception <class 'gdb.error'> There is no member named _M_dataplus.: 
spec=)
    at /home/kaveh/git/3.6/3rdParty/fuerte/src/ConnectionBuilder.cpp:195
#2  0x00000000015e92d2 in arangodb::network::ConnectionPool::selectConnection (this=this@entry=0x7ffff692e200, Python Exception <class 'gdb.error'> There is no member named _M_dataplus.: 
endpoint=, bucket=...)
    at /home/kaveh/git/3.6/arangod/Network/ConnectionPool.cpp:226
#3  0x00000000015e8ff2 in arangodb::network::ConnectionPool::leaseConnection (this=0x7ffff692e200, endpoint=...)
    at /home/kaveh/git/3.6/arangod/Network/ConnectionPool.cpp:62
#4  0x00000000015ec002 in arangodb::network::sendRequest (pool=pool@entry=0x7ffff692e200, dest=<incomplete type>, 
    type=type@entry=arangodb::fuerte::v1::RestVerb::Post, path=..., payload=..., options=..., 
    headers=std::map with 7886477447510829824 elements<error reading variable: Cannot access memory at address 0xd8>)
    at /home/kaveh/git/3.6/arangod/Network/Methods.cpp:133
```

Reply:
from `ConnectionBuilder::endpoint()`:
``` 
    throw std::logic_error("unsupported socket or protocol type");
      throw std::runtime_error(std::string("invalid socket type: ") +
    throw std::runtime_error(std::string("invalid schema: ") +
    throw std::runtime_error("invalid empty endpoint spec");
        throw std::runtime_error(std::string("invalid ipv6 address: ") + spec);
      throw std::runtime_error(std::string("invalid ipv6 address: ") + spec);
        throw std::runtime_error(std::string("invalid host in spec: ") + spec);
    throw std::runtime_error(std::string("invalid host: ") + spec);
        throw std::runtime_error(std::string("invalid port in spec: ") + spec);
      throw std::runtime_error(std::string("invalid port in spec: ") + spec);
```

Reply:
Obviously when creating a _new_ connection lots of things can go wrong, including OOM.